### PR TITLE
feat: Add select all shortcut to TextField and RichTextField

### PR DESCRIPTION
**Pull request by rhystmills**
## What does this change?

This adds a 'select all' shortcut to the TextField and RichTextField.

We filter the `baseKeymap` provided by `prosemirror-commands` here (based on key name) - removing the enter-based shortcuts which may interfere with intended behaviour in Composer.

| Before | After |
| ------  | ----- |
|  |  |

## How to test

Run `yarn start` from the root directory locally (with nginx running) to run the application.

The new cypress tests should also pass - run them with `yarn test:integration` while the application is running in another terminal.

## Accessibility

- [ ] [Tested with screen reader](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#screen-readers)
- [X] [Navigable with keyboard](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#keyboard-navigation)
- [ ] [Colour contrast passed](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#colour-contrast)
- [X] [The change doesn't use only colour to convey meaning](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#use-of-colour)
- [ ] [Interactive elements show a focus ring when focused by keyboard](https://github.com/guardian/accessibility/blob/main/editorial-tools-guidelines.md#appendix-the-focus-ring)
- [ ] [Semantic elements have been used where possible](https://github.com/guardian/accessibility/blob/main/editorial-tools-guidelines.md#appendix-semantic-html)

### CODE REVIEW

**Review by jonathonherbert — COMMENTED** [review]
Ace! Could we add an integration test to avoid regressions? E.g. select all in a text field, check that `win.navigator.clipboard.readText()` contains the relevant text?

(We can timebox this, so if this is a pain to test, we can just 🚢 !)

**Comment by rhystmills** [comment]
> Ace! Could we add an integration test to avoid regressions? E.g. select all in a text field, check that `win.navigator.clipboard.readText()` contains the relevant text?
> 
> (We can timebox this, so if this is a pain to test, we can just 🚢 !)

Good idea, I'll give it a go today.

File: cypress/tests/ImageElement.spec.ts [hunk 1]
```diff
@@ -224,13 +238,27 @@ describe("ImageElement", () => {
         setDocFromHtml(values);
         assertDocHtml(getSerialisedHtml(values));
       });
+
+      it.only("should select all of its text when the select all shortcut is used", () => {
```
Comment (jonathonherbert) [inline cypress/tests/ImageElement.spec.ts:242]: Don't forget to strip the `.only`s! Could have a CI linting rule, I do this all the time 😅

File: cypress/tests/ImageElement.spec.ts [hunk 2]
```diff
@@ -224,13 +238,27 @@ describe("ImageElement", () => {
...
     });
 
     describe("Checkbox field", () => {
       it(`should be clickable`, () => {
-        addImageElement();
-        getElementField("useSrc").find("input").click();
-        getElementField("useSrc").find("input").should("be.checked");
+        addImageElement({});
+        getElementRichTextField("useSrc").find("input").click();
+        getElementRichTextField("useSrc").find("input").should("be.checked");
```
Comment (jonathonherbert) [inline cypress/tests/ImageElement.spec.ts:261]: Is this change related?

File: cypress/tests/ImageElement.spec.ts (hunk 1 above)
Comment (rhystmills) [reply cypress/tests/ImageElement.spec.ts:242]: Whoops

File: cypress/tests/ImageElement.spec.ts (hunk 2 above)
Comment (rhystmills) [reply cypress/tests/ImageElement.spec.ts:261]: No - not sure how I managed that. Will fix.

**Review by jonathonherbert — APPROVED** [review]
